src/python/pants/jvm/types.py

```diff
@@ -0,0 +1,20 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
```

Comment:
Not sure how I feel about this file, it feels like an "other" file that you throw a bunch of unrelated things into. What's the motivation to move from `coursier_fetch.py`?

Reply:
I can give the module a better name, but these types are referenced in other files (besides `src/python/pants/jvm/resolve/coursier_fetch.py` and `src/python/pants/jvm/resolve/coursier_fetch_integration_test.py`):

```
src/python/pants/jvm/goals/coursier_integration_test.py
src/python/pants/backend/java/compile/javac_test.py
src/python/pants/backend/java/test/junit_test.py
```

I thought it made sense to have them live within their own file, as I prefer smaller files overall.

The other files are tests though and so are not production code. Is their a consensus on where these should live?

Reply:
In Python, this would probably be defined in `target_types.py` because it will soon be closely associated with the `JvmRequirement` target, right? We also have `PexRequirements` from `util_rules/pex.py`

Reply:
Somewhat associated. It represents the jar coordinate without referring to any target type / fields classes.

Reply:
I put the renamed types back into coursier_fetch for now. Seems like the consensus is to keep them in their existing module.